dask/array/core.py

```diff
@@ -2433,7 +2432,7 @@ def dot(a, b):
 
 def insert_to_ooc(out, arr, lock=True, region=None):
     if lock is True:
-        lock = Lock()
+        lock = SerializableLock()
```

Comment:
Previously it would have erred.  You're right though that now we're letting it succeed in a way that might be unsafe.